scripts/generate_topographic_data.py

```diff
@@ -0,0 +1,87 @@
+"""
+Script that takes the SRTM 30m worldwide elevation maps and creates a
+single representation of the area in NetCDF and GeoTIFF formats
+
+The SRTM files can be downloaded from NASA, and are in CRS EPSG:4326
+"""
+import os.path
+import glob
+import rasterio
+from rasterio.merge import merge
+from rasterio.warp import Resampling
+from nowcasting_dataset.geospatial import OSGB
+import rioxarray
+
+dst_crs = OSGB
+
+# Go through, open all the files, combined by coords, then save out to NetCDF, or GeoTIFF
+files = glob.glob("/SRTM/*.tif")
+out_dir = "/SRTM_TEMP/"
+
+
+upscale_factor = 0.12  # 30m to 250m-ish, just making it small enough files to actually merge
+for f in files:
+    with rasterio.open(f) as dataset:
+
+        # resample data to target shape
+        data = dataset.read(
+            out_shape=(
+                dataset.count,
+                int(dataset.height * upscale_factor),
+                int(dataset.width * upscale_factor),
+            ),
+            resampling=Resampling.bilinear,
+        )
+
+        # scale image transform
+        transform = dataset.transform * dataset.transform.scale(
+            (dataset.width / data.shape[-1]), (dataset.height / data.shape[-2])
+        )
+        name = f.split("/")[-1]
+        # Set the nodata values to 0, as nearly all ocean.
+        out_meta = dataset.meta.copy()
+        out_meta.update(
+            {
+                "driver": "GTiff",
+                "height": data.shape[1],
+                "width": data.shape[2],
+                "transform": transform,
+            }
+        )
+        with rasterio.open(os.path.join(out_dir, name), "w", **out_meta) as dest:
+            dest.write(data)
+files = glob.glob("/SRTM_TEMP/*.tif")
+src = rasterio.open(files[0])
+
+mosaic, out_trans = merge(files)
+out_meta = src.meta.copy()
+out_meta.update(
+    {
+        "driver": "GTiff",
+        "height": mosaic.shape[1],
+        "width": mosaic.shape[2],
+        "transform": out_trans,
+    }
+)
+out_fp = "europe_dem_250m.tif"
+with rasterio.open(out_fp, "w", **out_meta) as dest:
+    dest.write(mosaic)
+
+xds = rioxarray.open_rasterio(out_fp, parse_coordinates=True)
+# Reproject to exactly 1km pixels now
+with rasterio.open(out_fp) as src:
+    src_crs = src.crs
+    xds.attrs["crs"] = src_crs
+
+xds_resampled = xds.rio.reproject(dst_crs=dst_crs, resolution=500, resampling=Resampling.bilinear)
```

Comment:
Sorry if I've missed something but where does the `500` in `resolution=500` come from?  Half a kilometre per pixel?

Reply:
Cool beans.  It might be worth commenting on what the 500 means in the code?  No big worries if not though!

Reply:
Added it!